### PR TITLE
More imaging available in pipeline

### PR DESCRIPTION
**Pull request by FredSchuller**
Main changes are in function stx_imaging_pipeline, which now accepts an optional keyword 'method' to choose the imaging algorithm between MEM_GE (default), EM, and CLEAN.
In addition, there are a few minor (cosmetic) changes in several places.

### CODE REVIEW

**Comment by FredSchuller** [comment]
@samaloney @grazwegian 
can you please review this PR when you find some time?...

**Review by samaloney — APPROVED** [review]
Looks good to me. Does it need to be updated since #171 was merged?

File: stix/idl/processing/aux_data/stx_create_auxiliary_data.pro [hunk 1]
```diff
@@ -158,6 +160,11 @@ ROT_PITCH = -YAW * sin(ROLL_ANGLE * !dtor) + PITCH * cos(ROLL_ANGLE * !dtor)
 readcol, loc_file( 'Mapcenter_correction_factors.csv', path = getenv('STX_SAS') ), $
   avg_shift_x, avg_shift_y, offset_x, offset_y, /silent
 
+ ;;;;;;;
+ ; 2023-07-27: reset (avg_shift_x, avg_shift_y) to (0,0) in order to measure it again
+ ; with a larger sample of events
+ ;; avg_shift_x = 0.  &  avg_shift_y = 0.   ; commented out 2023-08-31
+
```
[the remark this answers is not in the document]
Comment (FredSchuller) [reply stix/idl/processing/aux_data/stx_create_auxiliary_data.pro]: removed in new commit

**Comment by FredSchuller** [comment]
@paolomassa @grazwegian 
could you please review this PR? thanks!

File: .gitignore
Comment (grazwegian) [inline .gitignore:1]: Not sure this should go into main repository

Comment (FredSchuller) [reply .gitignore:1]: Sure, I have now removed and stashed these

File: stix/idl/processing/imaging/fit_map_2dgauss.pro
Comment (paolomassa) [inline stix/idl/processing/imaging/fit_map_2dgauss.pro:1]: Do `gauss_X` and `gauss_Y` contain the X and Y coordinates of center of the Gaussian function in helioprojective Cartesian coordinates? I think that a comment on this should be added in the "OPTIONAL OUTPUTS" section of the header

File: stix/idl/processing/imaging/stx_imaging_pipeline.pro [hunk 2]
```diff
@@ -104,14 +145,39 @@ function stx_imaging_pipeline, stix_uid, time_range, energy_range, bkg_uid=bkg_u
...
+      path_bkg_file=path_bkg_file, xy_flare=flare_loc)
+
+    case method of
+      "MEM": out_map = stx_mem_ge(vis,imsize,pixel,aux_data,total_flux=max(abs(vis.obsvis)), /silent)
+      "CLEAN": begin
+        default, w_clean, 1  ; 1 = uniform weighting, 0 = natural weighting
+        niter  = 100   ; Number of iterations
+        gain   = 0.1   ; Gain used in each clean iteration
+        nmap   = 10    ; Plot clean components and cleaned map every 10 iterations
+        beam_width = 10. ; clean components are convolved with this beam
```
Comment (paolomassa) [inline stix/idl/processing/imaging/stx_imaging_pipeline.pro:172]: I think that the user should have the possibility to set the CLEAN beam width value. What about adding a keyword 'clean_beam_width'? Also, if the considered detectors are 3-10, then a default value of the beam width of 10 arcsec is probably too small. Indeed, the resolution of the finest detectors (detectors 3a, 3b and 3c) is ~14 arcsec

File: stix/idl/processing/imaging/stx_imaging_pipeline.pro [hunk 3]
```diff
@@ -104,14 +145,39 @@ function stx_imaging_pipeline, stix_uid, time_range, energy_range, bkg_uid=bkg_u
...
+        clean_map=stx_vis_clean(vis, aux_data, niter=niter, image_dim=imsize[0], PIXEL=pixel[0], $
+                                uniform_weighting = w_clean, gain=gain, nmap=nmap, $
+                                /plot, set_clean_boxes = 0, beam_width=beam_width)
```
Comment (paolomassa) [inline stix/idl/processing/imaging/stx_imaging_pipeline.pro:175]: I would give the user the possibility to set the clean boxes

File: .gitignore
Comment (paolomassa) [inline .gitignore:1]: Is it necessary to commit this file?

**Review by paolomassa — CHANGES_REQUESTED** [review]
See comments about 
1) gauss_X and gauss_Y coordinates
2) Clean beam width
3) Clean boxes
4) .gitignore file

File: stix/idl/processing/imaging/fit_map_2dgauss.pro
Comment (FredSchuller) [reply stix/idl/processing/imaging/fit_map_2dgauss.pro:1]: Yes, done.

File: stix/idl/processing/imaging/stx_imaging_pipeline.pro (hunk 2 above)
Comment (FredSchuller) [reply stix/idl/processing/imaging/stx_imaging_pipeline.pro]: OK, done.

File: stix/idl/processing/imaging/stx_imaging_pipeline.pro (hunk 3 above)
Comment (FredSchuller) [reply stix/idl/processing/imaging/stx_imaging_pipeline.pro]: OK, done.

**Review by paolomassa — APPROVED** [review]

**Review by paolomassa — APPROVED** [review]

**Pull request merged**